src/GetTaxonomiesDynamicFunctionReturnTypeExtension.php

```diff
@@ -33,27 +31,23 @@ public function getTypeFromFunctionCall(FunctionReflection $functionReflection,
     {
         // Called without second $output arguments
         if (count($functionCall->args) <= 1) {
-            return new ArrayType(new IntegerType(), new StringType());
+            return new ArrayType(new StringType(), new StringType());
         }
 
         $argumentType = $scope->getType($functionCall->args[1]->value);
 
         // When called with a non-string $output, return default return type
         if (! $argumentType instanceof ConstantStringType) {
-            return ParametersAcceptorSelector::selectFromArgs(
-                $scope,
-                $functionCall->args,
-                $functionReflection->getVariants()
-            )->getReturnType();
+            return new ArrayType(new StringType(), new ObjectType('WP_Taxonomy'));
```

Comment:
@johnbillion My problem is the following. When `get_taxonomies()` is given a variable PHPStan should let the return type unchanged.
What do you think?

Reply:
`@return array<string, string|\WP_Taxonomy>`

Reply:
Or is it `@return array<string|\WP_Taxonomy>`?

Reply:
As fas as I can tell, `get_taxonomies()` returns `array<string,\WP_Taxonomy>` for any value of `$output` except for `'names'` when it returns `array<string,string>`. So it's always possible to determine the exact return type.

Reply:
> So it's always possible to determine the exact return type.

When static analysis cannot figure out the parameter type - as it is a variable, not a constant - we are not able to predict the return type. Do you agree?